### PR TITLE
[move source lang] Type system rework. Cleanup HLIR/CFGIR

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Removed Base/Single type from all ASTs but HLIR and CFGIR
- There is now just 'Type'
- The structure imposed via the AST is now checked via constraints
- This greatly improves type inference in rough spots (locals, errors, and divergent expressions)
- HLIR and CFGIR now share types, commands, and expressions
- This greatly reduces the complexity of translation and code duplication 
    - In some spots there are some minimal asserts but I think its worth it

## Test Plan

- cargo test

### CODE REVIEW

**Comment by shazqadeer** [comment]
This is a big PR with changes in many source files.  The bulk of the 114 changed files are test files though.  The changes all seem to make sense.  I am happy to approve.  But it would be good if somebody more familiar with move-lang also took a look.

File: language/move-lang/src/typing/core.rs [hunk 1]
```diff
@@ -842,40 +892,52 @@ fn solve_builtin_type_constraint(
...
-    sp(loc, fs_)
+fn solve_single_type_constraint(context: &mut Context, loc: Loc, msg: String, ty: &Type) {
+    use TypeName_::*;
+    use Type_::*;
+    match unfold_type(&context.subst, ty.clone()).value {
+        Var(_) => unreachable!(),
+        Unit | Apply(_, sp!(_, Multiple(_)), _) => {
+            let tystr = error_format(ty, &context.subst);
+            let tmsg = format!(
+                "Expected a single type, but found expression list type: {}",
```
[the remark this answers is not in the document]
Comment (bob-wilson) [reply language/move-lang/src/typing/core.rs]: Oh, I guess this is probably intentional: Unit is "()", which is just a list type with 0 types. Right?

[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/src/typing/core.rs]: That's the idea, yeah.

These error messages will probably need to be tweaked at some point

**Pull request closed without merging**